### PR TITLE
Allow passing of custom arguments to urllib Request object

### LINKED ISSUE #33: Add support for passing custom urllib.request.Request object
This would allow for far more granular control of headers, proxies, etc. while not having to implement this functionality in pySmartDL. ~~Would close #30.~~ Sorry, misread documentation.

### PR DESCRIPTION
**Pull request by DashLt**
This will allow for more granular control of headers, proxies, and anything else that urllib provides an option for without having to implement the functionality in pySmartDL. Closes #33

### CODE REVIEW

File: pySmartDL/pySmartDL.py [hunk 1]
```diff
@@ -49,7 +49,9 @@ class SmartDL:
     The main SmartDL class
     
     :param urls: Download url. It is possible to pass unsafe and unicode characters. You can also pass a list of urls, and those will be used as mirrors.
-    :type urls: string or list of strings
+    :type urls: string or list of string
+    :param requestArgs: Arguments to be passed to a new urllib.request.Request instance in dictionary form. See https://docs.python.org/3/library/urllib.request.html#urllib.request.Request for options.
```
Comment (iTaybb) [inline pySmartDL/pySmartDL.py]: Please use the standard format for URLs (like in line 57)

File: pySmartDL/download.py [hunk 2]
```diff
@@ -4,16 +4,13 @@
 
 from . import utils
 
-def download(url, dest, startByte=0, endByte=None, headers=None, timeout=4, shared_var=None, thread_shared_cmds=None, logger=None, retries=3):
+def download(url, dest, requestArgs, startByte=0, endByte=None, timeout=4, shared_var=None, thread_shared_cmds=None, logger=None, retries=3):
```
Comment (iTaybb) [inline pySmartDL/download.py]: That's true that func download() is an internal function, but still i would like it to be generic and support not passing requestArgs at all (defaulting to None)

File: pySmartDL/pySmartDL.py [hunk 3]
```diff
@@ -75,14 +77,21 @@ class SmartDL:
             * If no path is provided, `%TEMP%/pySmartDL/` will be used.
     '''
     
-    def __init__(self, urls, dest=None, progress_bar=True, fix_urls=True, threads=5, timeout=5, logger=None, connect_default_logger=False):
+    def __init__(self, urls, dest=None, progress_bar=True, fix_urls=True, threads=5, timeout=5, logger=None, connect_default_logger=False, requestArgs=None):
```
Comment (iTaybb) [inline pySmartDL/pySmartDL.py]: 1. this is user-facing API and therefore requestArgs should become request_args
2. I feel like it should come after "fix_urls" or something, and not at the end of the constructor

Comment (DashLt) [reply pySmartDL/pySmartDL.py]: 1 and 3 have been fixed. As for 2, I put it at the end as to not break any code that currrently uses positional arguments. I can change it if you'd like though.

**Comment by iTaybb** [comment]
Thanks

**Pull request merged**